building/build-debs/homeworld-admin-tools/src/access.py

```diff
@@ -74,9 +74,17 @@ def access_ssh(add_to_agent=False):
     print("===== ^ CERTIFICATE DETAILS ^ =====")
     if add_to_agent:
         # TODO: clear old identities
-        if subprocess.call(["ssh-add", "--", keypath]) != 0:
-            fail_hint = "ssh-add returned non-zero exit code. do you have a ssh-agent?\n" \
-                "or, perhaps your agent is broken; consider killing it and launching a new one."
+        try:
+            ssh_add_output = subprocess.check_output(["ssh-add", "--", keypath], stderr=subprocess.STDOUT)
+            # if the user is using gnome, gnome-keyring might
+            # masquerade as ssh-agent and provide a zero exit
+            # code despite failing to add the certificate
+            if "add failed".encode() in ssh_add_output:
```

Comment:
Oh, wait, I just realized what you're doing.

You know you can write this as `b"add failed"` instead of `"add failed".encode()`, right?

Reply:
Actually.... I didn't :P Thanks, will change.